src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksWebServiceCallsServiceImpl.java

```diff
@@ -40,39 +44,68 @@
     public List<String> obtainPmwIdentifiersForPendingNewVendorRequests() {
         LOG.info("obtainPmwIdentifiersForPendingNewVendorRequests: Processing started.");
         List<String> pmwNewVendorIdentifiers = new ArrayList<String>();
-        List<PaymentWorksNewVendorRequestDTO> paymentWorksNewVendorRequestDTOs = retrieveAllPaymentWorksPendingNewVendorRequests(buildPaymentWorksPendingNewVendorRequestsURI());
+        List<PaymentWorksNewVendorRequestDTO> paymentWorksNewVendorRequestDTOs = retrieveAllPaymentWorksPendingNewVendorRequests();
         pmwNewVendorIdentifiers = getAllPaymentWorksIdentifiersFromDTO(paymentWorksNewVendorRequestDTOs);
         LOG.info("obtainPmwIdentifiersForPendingNewVendorRequests: Processing completed.");
         return pmwNewVendorIdentifiers;
     }
     
-    private List<PaymentWorksNewVendorRequestDTO> retrieveAllPaymentWorksPendingNewVendorRequests(URI pendingNewVendorsURI) {
-        LOG.info("retrieveAllPaymentWorksPendingNewVendorRequests: pendingNewVendorsURI=" + pendingNewVendorsURI);
+    private List<PaymentWorksNewVendorRequestDTO> retrieveAllPaymentWorksPendingNewVendorRequests() {
+        Client clientForNewVendorRequestsRootResults = null;
         Response responseForNewVendorRequestsRootResults = null;
         List<PaymentWorksNewVendorRequestDTO> pmwNewVendorIdentifiers = new ArrayList<PaymentWorksNewVendorRequestDTO>();
+        
         try{
-            responseForNewVendorRequestsRootResults = buildXmlOutput(pendingNewVendorsURI);
+            clientForNewVendorRequestsRootResults = constructClientToUseForPagedResponses();
+            responseForNewVendorRequestsRootResults = constructXmlResponseToUseForPagedData(clientForNewVendorRequestsRootResults, buildPaymentWorksPendingNewVendorRequestsURI());
             PaymentWorksNewVendorRequestsRootDTO newVendorsRoot = responseForNewVendorRequestsRootResults.readEntity(PaymentWorksNewVendorRequestsRootDTO.class);
             LOG.info("retrieveAllPaymentWorksPendingNewVendorRequests: newVendorsRoot.getCount()=" + newVendorsRoot.getCount());
-            if (newVendorsRoot.getCount() > 0) {
+            
+            while (thereAreMorePmwVendorIdsToRetrieve(newVendorsRoot)) {
                 pmwNewVendorIdentifiers.addAll(newVendorsRoot.getPmwNewVendorRequestsDTO().getPmwNewVendorRequests()); 
+                closeResponseJustObtained(responseForNewVendorRequestsRootResults);
+                
+                if (additionalPagesOfPmwVendorIdsExist(newVendorsRoot)) {
+                    responseForNewVendorRequestsRootResults = constructXmlResponseToUseForPagedData(clientForNewVendorRequestsRootResults, buildURI(newVendorsRoot.getNext()));
+                    newVendorsRoot = responseForNewVendorRequestsRootResults.readEntity(PaymentWorksNewVendorRequestsRootDTO.class);
+                }
+                else {
```

Comment:
Another "else" block start that could use adjustment.

Reply:
Moved